benchmarks/linear-elastic-plate-with-hole/parameter_extractor.py

```diff
@@ -0,0 +1,101 @@
+        file_name = os.path.basename(file_path)
+        if (
+            file_name.startswith("summary")
+            and rule_name == "summary"
+        ):
+            with open(file_path) as f:
+                data = json.load(f)
+            for experiment in data:
+                config = experiment['configuration']
+                results.setdefault(config, {})
+                results[config].setdefault('has parameter', [])
+                results[config].setdefault('investigates', [])
+                for key, val in experiment['parameters'].items():
+                    if isinstance(val, dict):
+                        results[config]["has parameter"].append({key: {
+                            "value": val["value"],
+                            "unit": self._get_unit(key),
+                            "json-path": f"/{key}/value",
+                            "data-type": self._get_type(val["value"]),
+                        }})
+                    else:
+                        results[config]["has parameter"].append({key: {
+                            "value": val,
+                            "unit": None,
+                            "json-path": f"/{key}",
+                            "data-type": self._get_type(val),
+                        }})
+                for key, val in experiment['metrics'].items():
+                    if isinstance(val, dict):
+                        results[config]["investigates"].append({key: {
+                            "value": val["value"],
+                            "unit": self._get_unit(key),
+                            "json-path": f"/{key}/value",
+                            "data-type": self._get_type(val["value"]),
+                        }})
+                    else:
+                        results[config]["investigates"].append({key: {
+                            "value": val,
+                            "unit": None,
+                            "json-path": f"/{key}",
+                            "data-type": self._get_type(val),
+                        }})        
+        return results
+
+    def extract_tools(self, rule_name: str, env_file_content: str,) -> dict:
+        results = {}
+        parsed = yaml.safe_load(env_file_content)
+        dependencies = parsed.get("dependencies", [])
+
+        for dep in dependencies:
+           if isinstance(dep, str):
+               match = re.match(r'^([a-zA-Z0-9_\-]+)([=<>!].*)?$', dep)
+               if match:
+                   name, version = match.groups()
+                   results[name] = version if version else None
+           elif isinstance(dep, dict):
+               for _, pkgs in dep.items():
+                   for pkg in pkgs:
+                       match = re.match(r'^([a-zA-Z0-9_\-]+)([=<>!].*)?$', pkg)
+                       if match:
+                           name, version = match.groups()
+                           results[name] = version if version else None
+        rename_map = {
```

Comment:
We have still not fully defined how we reference the software correctly, so I would avoid manual renaming here (we might at some point get confused why the names have changed), so just keep the package names as defined in the conda file.

Reply:
I fully appreciate that idea, though I'm not sure if we should to the mapping here, or in the knowledge graph.